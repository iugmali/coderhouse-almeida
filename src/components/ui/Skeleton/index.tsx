import Image from "next/image";
import {jetBrainsMono} from "@/app/fonts";

const shimmer =
  'before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:bg-gradient-to-r before:from-transparent before:via-white/60 before:to-transparent';

const ItemSkeleton = () => {
  return (
    <article
      className={`group rounded-3xl justify-self-center w-[288px] h-[350px] text-gray-950 shadow-sm hover:shadow-2xl flex flex-col justify-center items-center relative`}>
      <div className={`${shimmer} overflow-hidden relative h-[200px] w-[200px] bg-gray-100 rounded-3xl drop-shadow-sm`}>
        <Image className={`object-contain opacity-20`}
               src={"/image-placeholder.png"} alt={"produto sendo carregado..."} fill={true}
               sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw"
        />
      </div>
      <div className={`${shimmer} overflow-hidden relative w-60 bg-gray-100 h-6 mt-4 text-center text-xl py-2 rounded-3xl`} />
    </article>
  );
}

export const ItemListSkeleton = () => {
  return (
    <main className={`flex flex-col mt-4 mb-4`}>
      <h1 className={`${jetBrainsMono.className} text-center text-gray-600 text-sm md:text-base`}>Carregando...</h1>
      <section className={`flex-1 grid max-md:grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-8 mt-4 mx-auto`}>
        <ItemSkeleton/>
        <ItemSkeleton/>
        <ItemSkeleton/>
        <ItemSkeleton/>
        <ItemSkeleton/>
        <ItemSkeleton/>
      </section>
    </main>

  );
}
