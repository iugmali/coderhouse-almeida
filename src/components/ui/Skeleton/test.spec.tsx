import {render} from "@testing-library/react";
import {ItemListSkeleton} from "@/components/ui/Skeleton/index";

describe('Skeleton', () => {
  it('renders unchanged', () => {
    const {container} = render(<ItemListSkeleton />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
