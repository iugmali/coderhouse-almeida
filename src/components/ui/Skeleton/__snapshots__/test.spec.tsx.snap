// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton renders unchanged 1`] = `
<main
  class="flex flex-col mt-4 mb-4"
>
  <h1
    class="className text-center text-gray-600 text-sm md:text-base"
  >
    Carregando...
  </h1>
  <section
    class="flex-1 grid max-md:grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-8 mt-4 mx-auto"
  >
    <article
      class="group rounded-3xl justify-self-center w-[288px] h-[350px] text-gray-950 shadow-sm hover:shadow-2xl flex flex-col justify-center items-center relative"
    >
      <div
        class="before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:bg-gradient-to-r before:from-transparent before:via-white/60 before:to-transparent overflow-hidden relative h-[200px] w-[200px] bg-gray-100 rounded-3xl drop-shadow-sm"
      >
        <img
          alt="produto sendo carregado..."
          class="object-contain opacity-20"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw"
          src="/_next/image?url=%2Fimage-placeholder.png&w=3840&q=75"
          srcset="/_next/image?url=%2Fimage-placeholder.png&w=256&q=75 256w, /_next/image?url=%2Fimage-placeholder.png&w=384&q=75 384w, /_next/image?url=%2Fimage-placeholder.png&w=640&q=75 640w, /_next/image?url=%2Fimage-placeholder.png&w=750&q=75 750w, /_next/image?url=%2Fimage-placeholder.png&w=828&q=75 828w, /_next/image?url=%2Fimage-placeholder.png&w=1080&q=75 1080w, /_next/image?url=%2Fimage-placeholder.png&w=1200&q=75 1200w, /_next/image?url=%2Fimage-placeholder.png&w=1920&q=75 1920w, /_next/image?url=%2Fimage-placeholder.png&w=2048&q=75 2048w, /_next/image?url=%2Fimage-placeholder.png&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
      </div>
      <div
        class="before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:bg-gradient-to-r before:from-transparent before:via-white/60 before:to-transparent overflow-hidden relative w-60 bg-gray-100 h-6 mt-4 text-center text-xl py-2 rounded-3xl"
      />
    </article>
    <article
      class="group rounded-3xl justify-self-center w-[288px] h-[350px] text-gray-950 shadow-sm hover:shadow-2xl flex flex-col justify-center items-center relative"
    >
      <div
        class="before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:bg-gradient-to-r before:from-transparent before:via-white/60 before:to-transparent overflow-hidden relative h-[200px] w-[200px] bg-gray-100 rounded-3xl drop-shadow-sm"
      >
        <img
          alt="produto sendo carregado..."
          class="object-contain opacity-20"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw"
          src="/_next/image?url=%2Fimage-placeholder.png&w=3840&q=75"
          srcset="/_next/image?url=%2Fimage-placeholder.png&w=256&q=75 256w, /_next/image?url=%2Fimage-placeholder.png&w=384&q=75 384w, /_next/image?url=%2Fimage-placeholder.png&w=640&q=75 640w, /_next/image?url=%2Fimage-placeholder.png&w=750&q=75 750w, /_next/image?url=%2Fimage-placeholder.png&w=828&q=75 828w, /_next/image?url=%2Fimage-placeholder.png&w=1080&q=75 1080w, /_next/image?url=%2Fimage-placeholder.png&w=1200&q=75 1200w, /_next/image?url=%2Fimage-placeholder.png&w=1920&q=75 1920w, /_next/image?url=%2Fimage-placeholder.png&w=2048&q=75 2048w, /_next/image?url=%2Fimage-placeholder.png&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
      </div>
      <div
        class="before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:bg-gradient-to-r before:from-transparent before:via-white/60 before:to-transparent overflow-hidden relative w-60 bg-gray-100 h-6 mt-4 text-center text-xl py-2 rounded-3xl"
      />
    </article>
    <article
      class="group rounded-3xl justify-self-center w-[288px] h-[350px] text-gray-950 shadow-sm hover:shadow-2xl flex flex-col justify-center items-center relative"
    >
      <div
        class="before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:bg-gradient-to-r before:from-transparent before:via-white/60 before:to-transparent overflow-hidden relative h-[200px] w-[200px] bg-gray-100 rounded-3xl drop-shadow-sm"
      >
        <img
          alt="produto sendo carregado..."
          class="object-contain opacity-20"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw"
          src="/_next/image?url=%2Fimage-placeholder.png&w=3840&q=75"
          srcset="/_next/image?url=%2Fimage-placeholder.png&w=256&q=75 256w, /_next/image?url=%2Fimage-placeholder.png&w=384&q=75 384w, /_next/image?url=%2Fimage-placeholder.png&w=640&q=75 640w, /_next/image?url=%2Fimage-placeholder.png&w=750&q=75 750w, /_next/image?url=%2Fimage-placeholder.png&w=828&q=75 828w, /_next/image?url=%2Fimage-placeholder.png&w=1080&q=75 1080w, /_next/image?url=%2Fimage-placeholder.png&w=1200&q=75 1200w, /_next/image?url=%2Fimage-placeholder.png&w=1920&q=75 1920w, /_next/image?url=%2Fimage-placeholder.png&w=2048&q=75 2048w, /_next/image?url=%2Fimage-placeholder.png&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
      </div>
      <div
        class="before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:bg-gradient-to-r before:from-transparent before:via-white/60 before:to-transparent overflow-hidden relative w-60 bg-gray-100 h-6 mt-4 text-center text-xl py-2 rounded-3xl"
      />
    </article>
    <article
      class="group rounded-3xl justify-self-center w-[288px] h-[350px] text-gray-950 shadow-sm hover:shadow-2xl flex flex-col justify-center items-center relative"
    >
      <div
        class="before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:bg-gradient-to-r before:from-transparent before:via-white/60 before:to-transparent overflow-hidden relative h-[200px] w-[200px] bg-gray-100 rounded-3xl drop-shadow-sm"
      >
        <img
          alt="produto sendo carregado..."
          class="object-contain opacity-20"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw"
          src="/_next/image?url=%2Fimage-placeholder.png&w=3840&q=75"
          srcset="/_next/image?url=%2Fimage-placeholder.png&w=256&q=75 256w, /_next/image?url=%2Fimage-placeholder.png&w=384&q=75 384w, /_next/image?url=%2Fimage-placeholder.png&w=640&q=75 640w, /_next/image?url=%2Fimage-placeholder.png&w=750&q=75 750w, /_next/image?url=%2Fimage-placeholder.png&w=828&q=75 828w, /_next/image?url=%2Fimage-placeholder.png&w=1080&q=75 1080w, /_next/image?url=%2Fimage-placeholder.png&w=1200&q=75 1200w, /_next/image?url=%2Fimage-placeholder.png&w=1920&q=75 1920w, /_next/image?url=%2Fimage-placeholder.png&w=2048&q=75 2048w, /_next/image?url=%2Fimage-placeholder.png&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
      </div>
      <div
        class="before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:bg-gradient-to-r before:from-transparent before:via-white/60 before:to-transparent overflow-hidden relative w-60 bg-gray-100 h-6 mt-4 text-center text-xl py-2 rounded-3xl"
      />
    </article>
    <article
      class="group rounded-3xl justify-self-center w-[288px] h-[350px] text-gray-950 shadow-sm hover:shadow-2xl flex flex-col justify-center items-center relative"
    >
      <div
        class="before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:bg-gradient-to-r before:from-transparent before:via-white/60 before:to-transparent overflow-hidden relative h-[200px] w-[200px] bg-gray-100 rounded-3xl drop-shadow-sm"
      >
        <img
          alt="produto sendo carregado..."
          class="object-contain opacity-20"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw"
          src="/_next/image?url=%2Fimage-placeholder.png&w=3840&q=75"
          srcset="/_next/image?url=%2Fimage-placeholder.png&w=256&q=75 256w, /_next/image?url=%2Fimage-placeholder.png&w=384&q=75 384w, /_next/image?url=%2Fimage-placeholder.png&w=640&q=75 640w, /_next/image?url=%2Fimage-placeholder.png&w=750&q=75 750w, /_next/image?url=%2Fimage-placeholder.png&w=828&q=75 828w, /_next/image?url=%2Fimage-placeholder.png&w=1080&q=75 1080w, /_next/image?url=%2Fimage-placeholder.png&w=1200&q=75 1200w, /_next/image?url=%2Fimage-placeholder.png&w=1920&q=75 1920w, /_next/image?url=%2Fimage-placeholder.png&w=2048&q=75 2048w, /_next/image?url=%2Fimage-placeholder.png&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
      </div>
      <div
        class="before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:bg-gradient-to-r before:from-transparent before:via-white/60 before:to-transparent overflow-hidden relative w-60 bg-gray-100 h-6 mt-4 text-center text-xl py-2 rounded-3xl"
      />
    </article>
    <article
      class="group rounded-3xl justify-self-center w-[288px] h-[350px] text-gray-950 shadow-sm hover:shadow-2xl flex flex-col justify-center items-center relative"
    >
      <div
        class="before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:bg-gradient-to-r before:from-transparent before:via-white/60 before:to-transparent overflow-hidden relative h-[200px] w-[200px] bg-gray-100 rounded-3xl drop-shadow-sm"
      >
        <img
          alt="produto sendo carregado..."
          class="object-contain opacity-20"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw"
          src="/_next/image?url=%2Fimage-placeholder.png&w=3840&q=75"
          srcset="/_next/image?url=%2Fimage-placeholder.png&w=256&q=75 256w, /_next/image?url=%2Fimage-placeholder.png&w=384&q=75 384w, /_next/image?url=%2Fimage-placeholder.png&w=640&q=75 640w, /_next/image?url=%2Fimage-placeholder.png&w=750&q=75 750w, /_next/image?url=%2Fimage-placeholder.png&w=828&q=75 828w, /_next/image?url=%2Fimage-placeholder.png&w=1080&q=75 1080w, /_next/image?url=%2Fimage-placeholder.png&w=1200&q=75 1200w, /_next/image?url=%2Fimage-placeholder.png&w=1920&q=75 1920w, /_next/image?url=%2Fimage-placeholder.png&w=2048&q=75 2048w, /_next/image?url=%2Fimage-placeholder.png&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
      </div>
      <div
        class="before:absolute before:inset-0 before:-translate-x-full before:animate-[shimmer_2s_infinite] before:bg-gradient-to-r before:from-transparent before:via-white/60 before:to-transparent overflow-hidden relative w-60 bg-gray-100 h-6 mt-4 text-center text-xl py-2 rounded-3xl"
      />
    </article>
  </section>
</main>
`;
